src/main/java/com/develhope/spring/vehicles/services/VehicleService.java

```diff
@@ -40,23 +39,31 @@ public VehicleSavedDTO create(VehicleCreatorDTO vehicleCreatorDTO) {
         return vehicleMapper.toSavedDTO(savedVehicle);
     }
 
-    //TODO Convertire autorizzazione
-    public VehicleCreatorDTO update(long userId, long vehicleId, VehicleCreatorDTO vehicleCreatorDTO) {
-        checkUserAuthorizationBy(userId);
-        Vehicle existingVehicle;
-        existingVehicle = vehicleMapper.toEntity(vehicleCreatorDTO);
-        existingVehicle.setId(vehicleId);
-        Vehicle updatedVehicle = vehicleRepository.save(existingVehicle);
-        return vehicleMapper.toCreatorDTO(updatedVehicle);
+    public VehicleReworkedDTO update(long vehicleId, VehicleCreatorDTO vehicleCreatorDTO) {
+        Vehicle vehicleToUpdate = findVehicleBy(vehicleId);
```

Comment:
vedo che hai usato quel findVehicleById che girava in service. Sinceramente non so perchè ci sia, forse ppotrà servirci dopo forse eliminarlo non è necessario, però comunque a sto punto in orElseTrhow fagli lanciare l'eccezione NoSuchELement, non lasciarlo vuoto

Reply:
Ci sta, così non devi ogni volta metterti a scrivere tutta la stringa dell'optional e bla bla bla